# 표준 입력과 표준 출력

In [1]:
12345
10
#ipython 기본 기능
#print(12345) #표준 출력 기능을 내장하고 있음

10

In [2]:
print(12345)
10 #정수형 -> int

12345


10

In [3]:
'12345' #문자열 -> str

'12345'

In [4]:
type(12345)

int

In [5]:
type("12345")

str

In [6]:
type('12345')

str

In [7]:
"12345"*3

'123451234512345'

In [8]:
"num = "+"12345..."

'num = 12345...'

In [9]:
#값 입력 받기
name = input("당신의 이름을 입력하세요: ")
age = input("당신의 나이를 입력하세요: ")
# age = int(input("당신의 나이를 입력하세요: "))

당신의 이름을 입력하세요:  신윤서 
당신의 나이를 입력하세요:  15


In [10]:
print(name, age)

신윤서  15


In [11]:
print("나는",name,"이고 나이는", age, "입니다.")

나는 신윤서  이고 나이는 15 입니다.


In [12]:
print("나는 "+name+"이고 나이는 "+age+"입니다.")

나는 신윤서 이고 나이는 15입니다.


In [13]:
age=int(age)

In [14]:
age + 10

25

# 연습

In [15]:
# 자기 이름으로 삼행시 짓기

print("신: 신기하게 오늘")
print("윤: 윤슬도 봤고")
print("서: 서늘한 바람도 불잖아! 완벽!!")



신: 신기하게 오늘
윤: 윤슬도 봤고
서: 서늘한 바람도 불잖아! 완벽!!


In [16]:
first_line = input("신: ")
sec_line = input("윤: ")
thrid_line = input("서: ")

print(first_line, sec_line, thrid_line, sep="\n ")

신:  신기하게 오늘
윤:  윤슬도 봤고
서:  서늘한 바람도 불잖아! 완벽!!


신기하게 오늘
 윤슬도 봤고
 서늘한 바람도 불잖아! 완벽!!


문자열의 시작이 \ -> 이스케이프 문자, 특수문자 \n \\ \t

In [17]:
삼행시 = first_line + "\n"+ sec_line +"\n"+ thrid_line
print(삼행시)

신기하게 오늘
윤슬도 봤고
서늘한 바람도 불잖아! 완벽!!


# 변수

변수는 type safe: 변수를 선언할 때 타입(유형)을 지정하지 않아도 되는 특성   
변수양 = 값

## 변수의 명명규칙
1. 문자, 숫자, 밑줄을 가지고 만든다
2. 숫자로 시작하면 X
3. 대소문자를 구분
4. 특수기호, 공백 X

In [18]:
x = 10
y = 'hello'
z = 3.141592
x,y,z

(10, 'hello', 3.141592)

In [19]:
type(x),type(y),type(z)

(int, str, float)

In [20]:
print(type(x),type(y),type(z))

<class 'int'> <class 'str'> <class 'float'>


In [21]:
a,b,c = 10,20,30
a,b,c

(10, 20, 30)

In [22]:
a =10; b=20; c=30;

In [23]:
a = b= c= 40

In [24]:
a,b,c

(40, 40, 40)

In [25]:
d = True #bool
e = None #nonetype 
d,e

(True, None)

In [26]:
aa = 10
bb = aa
print(aa,bb)
id(aa), id(bb)

10 10


(140728258169560, 140728258169560)

In [27]:
a = 20
print(a,b)

20 40


In [29]:
#의미부여하기 어려운 변수명 > 수식이 아닌 경우는 사용하지 않음

#의미 있는 변수명

#대문자 upper-case 상수에 사용 
PI = 3.141592
MAX_CONNECTIONS = 10 #no. max connection
MIN_VALUE = 0 #min

# NAME_SPACE

변수가 최초로 정의된 영역에 따라 유효범위가 결정됩니다.

- 지역변수: 함수 내에서 선언
- 전역변수: 함수 영역 밖에서 선언한 변수, global 키워드
- 정적변수ㅣ클래스에서 사용하는 변수

In [33]:
#지역변수 
def local_example():
    local_var = 100
    print("여기는 함수의 영역입니다. ", local_var)

local_example()

여기는 함수의 영역입니다.  100


In [38]:
#전역변수
var_test = "나는 전역이다"
print(var_test)

def global_example():
    global var_test
    var_test = 100
    print("여기는 함수의 영역입니다. ", var_test)

global_example()
print(var_test)

나는 전역이다
여기는 함수의 영역입니다.  100
100


In [49]:
# 변수 영역 연습 

counter = 0 #global

def increment_counter():
    #counter = counter +1
    global counter
    counter += 1
    print("counter : ", counter)
    
increment_counter()
increment_counter()
increment_counter()

counter :  1
counter :  2
counter :  3


# 변수 연습

In [51]:
a = 1
b = "python"
c = [1, 2, 3]

a,b,c

(1, 'python', [1, 2, 3])

In [55]:
#02-8
a = [1, 2, 3]
b = a
print(a)
print(b)
a is b

a[1] = 4
print(a)
print(b)

[1, 2, 3]
[1, 2, 3]
[1, 4, 3]
[1, 4, 3]


In [69]:
#초보자를 위한 파이썬 300제
#파이썬 시작하기
#1
print("Hello World")
#2 
print("Mary's cosmetics")
#3
print('신씨가 소리질렀다. "도둑이야".')
#4
print("C:\Windows")
#5
print("안녕하세요. \n만나서\t반갑습니다.")
#6 
print("오늘은", "일요일")
#7 
print("naver", "kakao", "sk", "samsung", sep=";")
#8
print("naver", "kakao", "sk", "samsung", sep="/")
#9
print("first", end=""); print("second")

Hello World
Mary's cosmetics
신씨가 소리질렀다. "도둑이야".
C:\Windows
안녕하세요. 
만나서	반갑습니다.
오늘은 일요일
naver;kakao;sk;samsung
naver/kakao/sk/samsung
firstsecond


<>:9: SyntaxWarning: invalid escape sequence '\W'
<>:9: SyntaxWarning: invalid escape sequence '\W'
C:\Users\Admin\AppData\Local\Temp\ipykernel_6736\1678798798.py:9: SyntaxWarning: invalid escape sequence '\W'
  print("C:\Windows")


In [72]:
#파이썬 변수 
#11
삼성전자 = 50000
총_평가금액 = 삼성전자*10
print(총_평가금액)
#12
시가총액 = 298000000000000
현재가 = 50000
PER = 15.76
print(시가총액, type(시가총액))
print(현재가, type(현재가))
print(PER, type(PER))
#13
#14
#15
#16
#17              
#18
#19
#20


500000
298000000000000 <class 'int'>
50000 <class 'int'>
15.76 <class 'float'>


# 자료형

- 기본자료형: 숫자형, 문자열형 str,
- 복합 자료형 : 리스트, 셋, 튜플, 딕셔너리

|type|설명|예|
|-------|-------|-------|
|정수형 (int)| 정수를 저장하는 자료형 |1,2,-5|
|실수형 (float)|소수점이 있는 숫자를 저장하는 자료형|3.14, -0.001, 2.0|
|문자열 (str)|텍스트를 저장하는 자료형|"Hello", '파이썬'|
|논리형 (bool)|참/거짓 값을 저장하는 자료형|True, False|

In [74]:
#수치형 - int
20, type(20)

(20, int)

In [75]:
age = 10
type(age)

int

1. 진법: 10진수, 2진수, 8진수, 16진수
2. 입력하는 방법은 다르지만 print결과는 동일

In [80]:
a=10
b=0b0101 #2wlstn - 0/1fh vygusgksms qkdqjq -> 2^2+2^0
a,b

(10, 5)

In [81]:
c=0o10
d=0b1000
c,d

(8, 8)

In [82]:
#진수 리뷰

In [83]:
#float
b=3.15
b, type(b)

(3.15, float)

지수의 표기

숫자e2 -> 10^2의 곱 = 10**2
숫자e-2 -> 10^-2의 곱 = 10**-2

In [89]:
fa = 10e2
fa, 10**3

(1000.0, 1000)

In [88]:
fb = 10e-2
fb

0.1

In [90]:
fc = 3.1415e9
fc

3141500000.0

In [91]:
#boolean - True, False
#논리연산, 비교연산(T/F)에서 사용

In [93]:
ba = True
#bb = true -> 지정이 안되어있어서

In [94]:
# 논리연산자 - and or not

bc = 1
ba and bc

1

In [95]:
#사칙연산 - + - * / // %
a=2
b=3

In [98]:
a+b
a-b
a*b
a/b
a//b, a%b #몫, remainder


(0, 2)

In [106]:
#비교연산자
1+3 == 4
1+2 != 3
1+3 > 3
1+3 <= 4

True

In [108]:
#비교연산자와 논리연산자의 결합
10%2 == 0 and 12%3==0

True

In [110]:
a = '1234'
b = '5678'
a,b, id(a), id(b)
id(a) == id(b)

False

In [113]:
# 대입연산자
a = 10
b = a+2
a,b
a += 1
a

11

In [116]:
b= 34
b += 10 #44
b -= 5 #39
b

39

# 문자열 str

' ', " ", """ """, ''' ''' docstring

인코딩 > 코드 > 숫자값 정해져 있어요 "코드표"

unicode -> 문자 표준 
영어는 ascii -> 유니코드
한국어도 유니코드 

유니코드가 글자의 크디에 따라 ->  utf-8, [utf-16, utf-32]

파이썬의 문자열 str -> unicode base

euc-kr : 예전에 우리나라에서 쓰던 한글 코드

cp949 : MS windows의 한글코드

In [2]:
text = "파이썬"
print(text)
print(ord("파"))

파이썬
54028


In [15]:
#인코딩
enc_text = text.encode("utf-8")
enc_text

b'hi this is python'

In [16]:
dec_text = enc_text.decode("utf-8")
dec_text

'hi this is python'

In [17]:
text = "신윤서"

In [18]:
text="hi this is python"

In [117]:
a = "hello"
b = "python"
a+b

'hellopython'

In [118]:
c="*"
c*10

'**********'

In [119]:
e = '1'
e == '1'


True

In [127]:
#문자열은 리스트, 열거형
a = '20010331Rabcd' #list of 13 letters
a[0] #자릿수 ->index
a[3] #indexing
a[0:4] #영역을 지정할 때 : (콜론 사용) -> slicing
a[:4]
a[-5:-2] #a[8:11]
a[9:]

'abcd'

In [129]:
s = 'abc'
a = s
a is s

True

In [132]:
b = s[:]
b
a is b

True

In [24]:
doc_string = '''docstring \n
start \n
end \n
'''

""" """
#shift+tab -> 함수의 문자 열림 
print(doc_string)

docstring 

start 

end 




# str의 함수

dir()

In [25]:
dir(text)

['__add__',
 '__class__',
 '__contains__',
 '__delattr__',
 '__dir__',
 '__doc__',
 '__eq__',
 '__format__',
 '__ge__',
 '__getattribute__',
 '__getitem__',
 '__getnewargs__',
 '__getstate__',
 '__gt__',
 '__hash__',
 '__init__',
 '__init_subclass__',
 '__iter__',
 '__le__',
 '__len__',
 '__lt__',
 '__mod__',
 '__mul__',
 '__ne__',
 '__new__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 '__rmod__',
 '__rmul__',
 '__setattr__',
 '__sizeof__',
 '__str__',
 '__subclasshook__',
 'capitalize',
 'casefold',
 'center',
 'count',
 'encode',
 'endswith',
 'expandtabs',
 'find',
 'format',
 'format_map',
 'index',
 'isalnum',
 'isalpha',
 'isascii',
 'isdecimal',
 'isdigit',
 'isidentifier',
 'islower',
 'isnumeric',
 'isprintable',
 'isspace',
 'istitle',
 'isupper',
 'join',
 'ljust',
 'lower',
 'lstrip',
 'maketrans',
 'partition',
 'removeprefix',
 'removesuffix',
 'replace',
 'rfind',
 'rindex',
 'rjust',
 'rpartition',
 'rsplit',
 'rstrip',
 'split',
 'splitlines',
 'startswith',
 'stri

In [34]:
text.__add__("!")

'hi this is python!'

In [33]:
print(text)
text.count("p")

hi this is python


1

In [35]:
text.islower()

True

In [36]:
text.upper()

'HI THIS IS PYTHON'

In [37]:
text.split(" ")

['hi', 'this', 'is', 'python']

escape 문자
\

In [43]:
print("\'hello\nworld\' ") #escape 문자는 특수문자를 그냥 텍스트로 인식하게 함

'hello
world' 


raw-string : 모두 다 문자열일때 r'str?\'

f-string : 형식 코드가 내포된 문자열 f'str{text}'

In [48]:
tr_1 = "12\4" #'12\x04'
tr_1

'12\x04'

In [62]:
raw_str = r'tr_1'
raw_str

'tr_1'

f' {수식} 문자열이에요 {변수}'

In [67]:
name = "Iris"
age = 23

print(f"제 이름은 {name}입니다. 나이는 {age}살 입니다. ")

제 이름은 Iris입니다. 나이는 23살 입니다. 


In [70]:
name = input("이름을 입력하세요: ")
age = input("나이를 입력하세요: ")

output = f"안녕하세요, 제 이름은 {name}이고 나이는 {age}살 입니다! "
print(output)
        

이름을 입력하세요:  신윤서
나이를 입력하세요:  23


안녕하세요, 제 이름은 신윤서이고 나이는 23살 입니다! 


In [78]:
#dir(f_str_!)
format_str = ' {0:.4f} {1:s} are worth US$ {2:d}' #template
format_str.format(99.99, "dollars", 100)

' 99.9900 dollars are worth US$ 100'

In [82]:
menu = input("주문할 메뉴를 입력하세요: ")
price = int(input("한개의 가격이 얼마인가요? "))
count = int(input("몇 개나 주문하세요?"))

total = price*count
msg = '{} 메뉴의 {}개의 가격은 총 {:,}입니다. '

print(msg.format(menu, count, total))

주문할 메뉴를 입력하세요:  떡볶이
한개의 가격이 얼마인가요?  5000
몇 개나 주문하세요? 2


떡볶이 메뉴의 2개의 가격은 총 10,000입니다. 


In [107]:
items = [
    {"name": "아메리카노", "price": 3000, "qty": 2},
    {"name": "카페라떼", "price": 4000, "qty": 1},
    {"name": "샌드위치", "price": 5500, "qty": 3}
]

header_str = "{0:10}{1:10}{2:10}{3:10}" # 인덱스 데이터길이
print(header_str.format("상품명","가격","수량","총액"))
print("-"*50)

data_str = "{:10}{:10,}{}{:10}{:10,}{}"
for item in items : 
    item_name = item["name"]
    item_price = item["price"]
    item_qty = item["qty"]
    total_price = item_price*item_qty
    print(data_str.format(item_name,item_price,"원",item_qty,total_price,"원"))

    
#data_str = "{}{}{}{}"

상품명       가격        수량        총액        
--------------------------------------------------
아메리카노          3,000원         2     6,000원
카페라떼           4,000원         1     4,000원
샌드위치           5,500원         3    16,500원


TypeError, ValueError

In [54]:
b = 2
c = '3'
#b+c #typeerror
b + int(c)

5

In [61]:
c = '3a'
b + int(c[0]) #ValueError                                

5

| 함수                  | 설명                  | 예시                                        |
| ------------------- | ------------------- | ----------------------------------------- |
| lower()           | 모두 소문자로 변환          | "Python".lower() → "python"           |
| upper()           | 모두 대문자로 변환          | "python".upper() → "PYTHON"           |
| capitalize()      | 첫 글자만 대문자           | "hello".capitalize() → "Hello"        |
| title()           | 각 단어의 첫 글자 대문자      | "hello world".title() → "Hello World" |
| strip()           | 앞뒤 공백 제거            | " hello ".strip() → "hello"           |
| lstrip()          | 왼쪽 공백 제거            | " hello".lstrip()                       |
| rstrip()          | 오른쪽 공백 제거           | "hello ".rstrip()                       |
| replace(old, new) | 문자열 바꾸기             | "hello".replace("l", "x") → "hexxo"   |
| split()           | 문자열 나누기 (기본: 공백)    | "a b c".split() → ['a', 'b', 'c']     |
| join(list)        | 리스트를 문자열로 결합        | " ".join(['a', 'b']) → "a b"          |
| find(sub)         | 처음 등장하는 위치 (없으면 -1) | "apple".find("p") → 1                 |
| index(sub)        | 처음 등장하는 위치 (없으면 오류) | "apple".index("p") → 1                |
| count(sub)        | 부분 문자열 개수           | "banana".count("a") → 3               |
| startswith(sub)   | 해당 문자열로 시작?         | "python".startswith("py") → True      |
| endswith(sub)     | 해당 문자열로 끝남?         | "python".endswith("on") → True        |
| isalnum()         | 영문자+숫자인가?           | "abc123".isalnum() → True             |
| isalpha()         | 영문자만인가?             | "abc".isalpha() → True                |
| isdigit()         | 숫자만인가?              | "123".isdigit() → True                |
| isspace()         | 공백문자만 있는가?          | " \t\n".isspace() → True              |

In [114]:
text = "hello python!"
word = "python"
text_space = "     hello python!     "

print(text.upper())
print(text.lower())
print(word.capitalize())
print(text.capitalize())
print(text.title())
print(text_space.strip())
print(text_space.lstrip())
print(text_space.rstrip())
print(text_space.isalpha())
print(word.isalpha())

HELLO PYTHON!
hello python!
Python
Hello python!
Hello Python!
hello python!
hello python!     
     hello python!
False
True


# 문자열 함수 연습   
Q1. 홍길동 씨의 주민등록번호는 881120-1068234이다. 홍길동씨의 주민등록번호를 연월일 [YYYYMMDD] 부분과 그 뒤의 숫자 부분으로 나누어 출력해 보자.   

Q2. 주민등록번호 뒷자리의 맨 첫 번째 숫자는 성별을 나타낸다. 주민등록번호에서 성별을 나타내는 숫자를 출력해 보자

Q3. 숫자와 문자를 나누어 a1,a2 변수에 넣으세요.

a = '20010331Rainy'

Q4. 모두 대문자로, 모두 소문자로 바꾸세요.

a = 'Hi'

Q5. 아래 변수의 화이트 스페이스를 제거하고 출력하세요.

a = ' hi '

Q6. 아기상어 동요 속에는 '뚜루' 가 몇개나 들어 있을까요?

In [124]:
#Q1
id_num = ("881120-1068234")
id_lst = id_num.split("-")
print(id_lst)
#2
print(id_lst[1][0])

'1'

In [127]:
#Q3
a = '20010331Rainy'

#Q4
a = "Hi"
print(a.upper())
#Q5
a = ' hi '
print(a.strip())


HI
hi


In [4]:
#Q6
lyrics = '''아기상어 뚜루루뚜루
귀여운
뚜루루뚜루
바닷속
뚜루루뚜루
아기상어!

엄마상어
뚜루루뚜루
어여쁜
뚜루루뚜루
바닷속
뚜루루뚜루
엄마상어!

아빠상어
뚜루루뚜루
힘이 센
뚜루루뚜루
바닷속
뚜루루뚜루
아빠상어!

할머니상어
뚜루루뚜루
자상한
뚜루루뚜루
바닷속
뚜루루뚜루
할머니상어!

할아버지상어
뚜루루뚜루
멋있는
뚜루루뚜루
바닷속
뚜루루뚜루
할아버지상어!

우리는
뚜루루뚜루
바다의
뚜루루뚜루
사냥꾼
뚜루루뚜루
상어가족!



상어다
뚜루루뚜루
도망쳐
뚜루루뚜루
도망쳐
뚜루루뚜루
숨자! 으악!

한번 더!


상어다
뚜루루뚜루
도망쳐
뚜루루뚜루
도망쳐
뚜루루뚜루
숨자! 으악!

아기상어
뚜루루뚜루
귀여운
뚜루루뚜루
바닷속
뚜루루뚜루
아기상어!

엄마상어
뚜루루뚜루
어여쁜
뚜루루뚜루
바닷속
뚜루루뚜루
엄마상어!

아빠상어
뚜루루뚜루
힘이 센
뚜루루뚜루
바닷속
뚜루루뚜루
아빠상어!

할머니상어
뚜루루뚜루
자상한
뚜루루뚜루
바닷속
뚜루루뚜루
할머니상어!

할아버지상어
뚜루루뚜루
멋있는
뚜루루뚜루
바닷속
뚜루루뚜루
할아버지상어!

우리는
뚜루루뚜루
바다의
뚜루루뚜루
사냥꾼
뚜루루뚜루
상어가족!


상어다
뚜루루뚜루
도망쳐
뚜루루뚜루
도망쳐
뚜루루뚜루
숨자! 으악!

살았다
뚜루루뚜루
살았다
뚜루루뚜루
오늘도
뚜루루뚜루
살았다!

신난다
뚜루루뚜루
신난다
뚜루루뚜루
춤을 춰
뚜루루뚜루
노래 끝!'''

print(lyrics.count("뚜루"))

102
